lib/asciidoctor/opal_ext/file.rb

```diff
@@ -135,6 +135,46 @@ def self.read(path)
         }
         return data;
       )
+    # afx means asciidocfx
+    when 'javafx'
+      %x(
+        var data = '';
+
+        if(afx) {
+          try {
+            data = afx.readAsciidoctorResource(path,null);
+            if(data==404)
+            throw new Error();
+          }
+          catch (e) {
+            throw #{IOError.new `'No such file or directory: ' + path`};
+          }
+          return data;
+        }
+
+        var status = -1;
+        try {
+          var xhr = new XMLHttpRequest();
```

Comment:
I tried to implement one method for duplicated implementation for XHR request. But, I couldnt do it because of my lower experience in ruby.